File: backend/models/project.go
package models

import "time"

// Project represents a storyboard project
type Project struct {
	ID              int64      `json:"id"`
	Name            string     `json:"name"`
	Description     string     `json:"description"`
	ScriptText      string     `json:"script_text"`
	VideoURL        string     `json:"video_url"`
	VideoPreviewURL string     `json:"video_preview_url"`
	VideoStatus     string     `json:"video_status"`
	VideoError      string     `json:"video_error"`
	VideoDuration   float64    `json:"video_duration"`
	PinnedAt        *time.Time `json:"pinned_at,omitempty"`
	IsPinned        bool       `json:"is_pinned"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

// ProjectWithStats includes project with statistics
type ProjectWithStats struct {
	Project
	ChapterCount    int `json:"chapter_count"`
	SceneCount      int `json:"scene_count"`
	StoryboardCount int `json:"storyboard_count"`
}

File: storyboard-app/src/app/api/types/project.ts
export type Project = {
  id: number;
  name: string;
  description: string;
  script_text?: string;
  scriptText?: string;
  video_url?: string;
  video_preview_url?: string;
  video_poster_url?: string;
  video_status?: string;
  video_error?: string;
  video_duration?: number;
  chapter_count?: number;
  scene_count?: number;
  storyboard_count?: number;
  pinned_at?: string;
  is_pinned?: boolean;
  created_at?: string;
  updated_at?: string;
};

export type Chapter = {
  id: number;
  project_id: number;
  title: string;
  summary: string;
  sort_order: number;
  created_at?: string;
  updated_at?: string;
};

export type ComposeProjectVideoResult = {
  project_id: number;
  video_url: string;
  video_preview_url?: string;
  video_status?: string;
  video_error?: string;
  video_duration?: number;
  project: Project;
};
